src/libraries/Microsoft.Extensions.Caching.Memory/src/MemoryCache.cs

```diff
@@ -481,7 +477,7 @@ private void TriggerOvercapacityCompaction()
             // If there is already a thread that is running compact - do nothing
             if (Interlocked.CompareExchange(ref lockFlag, 1, 0) == 0)
                 // Spawn background thread for compaction
-                ThreadPool.QueueUserWorkItem(s =>
+                ThreadPool.UnsafeQueueUserWorkItem(s =>
```

Comment:
it looks like this still captures `this` for `lockFlag`; should `lockFlag` be accessed via `s` ? perhaps try marking `(static s => { ...})` ?